src/types.ts

```diff
@@ -243,3 +243,37 @@ export interface Organisation {
 export interface UserOrg extends User, Organisation {
   id: number
 }
+
+export interface ServerPreferences {
+  thumbnailMaxWidth: number
+  thumbnailMaxHeight: number
+  hoursSchedule?: number[]
+  SMTPConfig?: {
+    host: string
+    port: number
+    secure: boolean
+    user: string
+    defaultFromName: string
+    defaultFromEmail: string
+  }
+  systemManagerPermissionName?: string
+  previewDocsMinKeepTime?: string
+  previewDocsCleanupSchedule?: number[]
+  backupSchedule?: number[]
+  backupFilePrefix?: string
+  maxBackupDurationDays?: number
+}
+
+export const serverPrefKeys: (keyof ServerPreferences)[] = [
```

Comment:
I wanted the type of this array to be an *exhaustive* check of all pref keys, so it will warn if any are missing. Best I could do was do show error if incorrect ones are included, but couldn't get it to be exhaustive.